src/hotspot/share/gc/shenandoah/shenandoahParallelCleaning.cpp

```diff
@@ -31,22 +31,28 @@
 #include "gc/shenandoah/shenandoahParallelCleaning.hpp"
 #include "runtime/safepoint.hpp"
 
-ShenandoahClassUnloadingTask::ShenandoahClassUnloadingTask(BoolObjectClosure* is_alive,
+ShenandoahClassUnloadingTask::ShenandoahClassUnloadingTask(ShenandoahPhaseTimings::Phase phase,
+                                                           BoolObjectClosure* is_alive,
                                                            uint num_workers,
                                                            bool unloading_occurred) :
   AbstractGangTask("Shenandoah Class Unloading"),
+  _phase(phase),
   _unloading_occurred(unloading_occurred),
   _code_cache_task(num_workers, is_alive, unloading_occurred),
   _klass_cleaning_task() {
   assert(SafepointSynchronize::is_at_safepoint(), "Must be at a safepoint");
 }
 
 void ShenandoahClassUnloadingTask::work(uint worker_id) {
-  _code_cache_task.work(worker_id);
+  {
+    ShenandoahWorkerTimingsTracker x(_phase, ShenandoahPhaseTimings::CodeCacheRoots, worker_id);
+    _code_cache_task.work(worker_id);
```

Comment:
This does not look like "normal" code cache root operation, though, right? Consider adding another type to `SHENANDOAH_PAR_PHASE_DO` instead? I.e. `CodeCacheUnload`?